### ISSUE TITLE
"TypeError: 'ellipsis' object is not iterable" on python 3.5.2

### ISSUE THREAD

**Issue by ctz** [hamcrest/PyHamcrest#146]
Repro steps -- first `pip3 install pyhamcrest` -- i got pyhamcrest-2.0.2. Second:

```
$ python3
Python 3.5.2 (default, Apr 16 2020, 17:47:17) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import hamcrest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jbp/.local/lib/python3.5/site-packages/hamcrest/__init__.py", line 1, in <module>
    from hamcrest.core import *
  File "/home/jbp/.local/lib/python3.5/site-packages/hamcrest/core/__init__.py", line 2, in <module>
    from hamcrest.core.core import *
  File "/home/jbp/.local/lib/python3.5/site-packages/hamcrest/core/core/__init__.py", line 13, in <module>
    from hamcrest.core.core.raises import calling, raises
  File "/home/jbp/.local/lib/python3.5/site-packages/hamcrest/core/core/raises.py", line 15, in <module>
    class Raises(BaseMatcher[Callable[..., Any]]):
  File "/usr/lib/python3.5/typing.py", line 1025, in __getitem__
    tvars = _type_vars(params)
  File "/usr/lib/python3.5/typing.py", line 284, in _type_vars
    _get_type_vars(types, tvars)
  File "/usr/lib/python3.5/typing.py", line 279, in _get_type_vars
    t._get_type_vars(tvars)
  File "/usr/lib/python3.5/typing.py", line 786, in _get_type_vars
    _get_type_vars(self.__args__, tvars)
  File "/usr/lib/python3.5/typing.py", line 277, in _get_type_vars
    for t in types:
TypeError: 'ellipsis' object is not iterable
>>> 
```

This is ubuntu 16.04.6 LTS, where the very latest version of python available is this one -- 3.5.2 -- and no later versions are available.

**Comment by ctz** [comment]
I've gone back a few versions to find one that works:

- 2.0.1: not working
- 2.0.0: not working
- 1.10.1: working

So I'll stick with that version for now.

**Comment by brunns** [comment]
Seems to work under 3.5.9, so I think this is a Python bug (in the new at that time `typing` module) rather than a PyHamcrest bug, fixed somewhere along the 3.5 line.
